src/NuGetPackageVerifier/Rules/AssemblyHasCommitHashAttributeRule.cs

```diff
@@ -11,18 +11,20 @@
 
 namespace NuGetPackageVerifier.Rules
 {
-    public class AssemblyHasCommitHashAttributeRule : AssemblyHasAttributeRuleBase
+    public class AssemblyHasCommitHashAttributeRule : IPackageVerifierRule
     {
-        public override IEnumerable<PackageVerifierIssue> ValidateAttribute(
-            string currentFilePath,
-            AssemblyDefinition assembly,
-            Collection<CustomAttribute> assemblyAttributes)
+        public IEnumerable<PackageVerifierIssue> Validate(PackageAnalysisContext context)
         {
-            var fileName = Path.GetFileNameWithoutExtension(currentFilePath);
-            var isSourcesPackage = fileName.EndsWith(".Sources", StringComparison.OrdinalIgnoreCase);
-            if (!isSourcesPackage && !HasCommitHashInMetadataAttribute(assemblyAttributes))
+            AssemblyHasAttributeHelper.GetAssemblyAttributesData(context);
```

Comment:
That works too. The only disadvantage I can think of with that is we might populate `AssemblyData` even if no rules actually run on that. But it is a rare case and I don't know if we care enough about it. @Eilon thoughts?